addons/product/models/product_product.py

```diff
@@ -174,10 +174,10 @@ def init(self):
 
     @api.constrains('barcode')
     def _check_barcode_uniqueness(self):
-        """ With GS1 nomenclature, products and packagins use the same pattern. Therefore, we need
+        """ With GS1 nomenclature, products and packagings use the same pattern. Therefore, we need
         to ensure the uniqueness between products' barcodes and packagings' ones"""
-        condition = expression.OR([[('barcode', '=', b)] for b in self.mapped('barcode') if b])
-        if self.env['product.packaging'].search(condition, limit=1):
+        domain = [('barcode', 'in', [b for b in self.mapped('barcode') if b])]
```

Comment:
It's better to avoid the `mapped` in this case as it is useless, it isn't ?
```suggestion
        domain = [('barcode', 'in', [product.barcode for product in self if product.barcode])]
```
And the same change can be done for the packaging's barcode check too I guess :)

Reply:
IMO, Not really, it is the purpose of mapped. 
In the performance point of view, `mapped` is better because it avoids extra cache lookup (but it won't be significant).

:o I didn't see the other one, I will fix too.